Mathlib/RepresentationTheory/Basic.lean

```diff
@@ -621,21 +618,15 @@ theorem tprod_apply (g : G) : (ρV ⊗ ρW) g = TensorProduct.map (ρV g) (ρW g
   rfl
 
 theorem smul_tprod_one_asModule (r : MonoidAlgebra k G) (x : V) (y : W) :
-    -- Porting note: required to since Lean 4 doesn't unfold asModule
-    let x' : ρV.asModule := x
-    let z : (ρV.tprod 1).asModule := x ⊗ₜ y
-    r • z = (r • x') ⊗ₜ y := by
+    r • (show (ρV.tprod 1).asModule from x ⊗ₜ y) = (r • show ρV.asModule from x) ⊗ₜ y := by
```

Comment:
This should really be a rather short statement. But the `show blablabla from` (2x) makes it quite ugly...

Is this lemma maybe no longer the right thing to have?

Reply:
Lean 3 did use type ascriptions here: `(r • (x ⊗ₜ y) : (ρV.tprod 1).as_module) = (r • x : ρV.as_module) ⊗ₜ y`. Not sure that `show` is much worse.

More generally I feel like representation theory does a lot of defeq trickery, but I don't have enough feel for this corner of Mathlib to say where the exact fix should be. This lemma does get used in one place, Mathlib/RepresentationTheory/Homological/Resolution.lean. (And it works as `with_reducible by exact`, so apparently we do want it.)

Reply:
Aah, indeed `show` isn't much worse then.